src/ReactiveUI/ReactiveList.cs

```diff
@@ -579,7 +580,10 @@ public IDisposable SuppressChangeNotifications()
 
             return new CompositeDisposable(this.suppressChangeNotifications(), Disposable.Create(() => {
                 if (Interlocked.Decrement(ref _resetNotificationCount) == 0) {
-                    publishResetNotification();
+                    if (resetEventArgs == null) {
```

Comment:
Potentially a threading issue here, since your dispose may not be called in the same thread as when it was created. Probably should use a Interlocked.Exchange or something?

Reply:
It might be alright given that it's inside the Interlocked.Decrement() though.

Reply:
As `resetEventArgs` is set (if it is set) before the disposable is returned, and nothing can change its value until the disposable is called, I don't really see how a race condition could occur? Duplicate/missing changed notifications cannot occur since `Interlocked.Decrement` is still used. Am I missing something?

Reply:
Yeah I agree, was just a initial reaction until I saw everything inside the Decrement statement.